srml/support/src/error.rs

```diff
@@ -115,6 +116,29 @@ macro_rules! decl_error {
 				$crate::dispatch::DispatchError::new(None, self.as_u8(), Some(self.as_str()))
 			}
 		}
+
+		impl $crate::error::ModuleErrorMetadata for $error {
+			fn metadata() -> &'static [ $crate::error::ErrorMetadata] {
+				&[
+					$crate::error::ErrorMetadata {
```

Comment:
I'm not sure we want to add these inherent error descriptions every time.

What do you think @jacogr?

If we want them, we should add them only once.

Reply:
I am also not sure what is the best way to handle those. Have another field in `ModuleMetadata` can avoid the duplicates but just don't feel right to me. Or just omitting them and that means SDK have to hardcode those messages.
Also du-dupe those will mean every Error type must have `Other` and `CannotLookup` as the first two variants, which could be an acceptable trade-off.

Reply:
Not 100% sure on the scope, if it is just `Other` and `CannotLookup` and they have static indexes for all, I think it is quite acceptable to allow them to be coded externally. 

If we want to move these top-level, I'd rather revisit that we we do types as well (which needs top-level information).

The alternative (and bear in mind this is my first time looking), would probably be to just have all the errors messages top-level and each module errors just has an index to the message.

Can do either, but for V8, I would probably lean "to omitting, specified as defaults" instead of doing too much?

Reply:
PS: Extracting with the current branch, I only see the errors structure being non-empty for the system module, i.e. all the other modules are empty. (So don't see duplication atm)

Reply:
right now only system and balances module are using decl_error but I will expect balances module also have errors field. will take a look later

Reply:
Only system declares an error, so that is expected behavior. 

> The alternative (and bear in mind this is my first time looking), would probably be to just have all the errors messages top-level and each module errors just has an index to the message.

Hmm, would add too much complexity as we don't know error types of other modules in front of. I think the current approach is okay. Maybe at some point later when we need a 2 pass strategy anyway, we can change this as well.

`Other` and `CannotLookup` will be present for every error.

Reply:
I have removed those inherent errors from metadata so SDK will need to know the indices starts from 2